### PR TITLE
GEN-1386 - feat(widget): added foundation for CalculatePricePage

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

Adds the foundation for the addition of Widget Calculate Price page. 

## Justify why they are needed

Part of Widget feature. Observe that rendering PriceCalculator outside PDP is a trick task which will be tackled on subsequent PRs.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3534** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3534" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3534?utm_source=stack-comment).

File: apps/store/src/features/widget/CalculatePricePage.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { Space, Heading } from 'ui'
+import { Pillow } from '@/components/Pillow/Pillow'
+import { SpaceFlex } from '@/components/SpaceFlex/SpaceFlex'
+import { WidgetPriceIntentFragment } from '@/services/apollo/generated'
```
Comment (robinandeer) [inline apps/store/src/features/widget/CalculatePricePage.tsx:4]: nit

```suggestion
import { type WidgetPriceIntentFragment } from '@/services/apollo/generated'
```

File: apps/store/src/features/widget/CalculatePricePage.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { Template } from '@/services/PriceCalculator/PriceCalculator.types'
```
Comment (robinandeer) [inline apps/store/src/features/widget/CalculatePricePage.tsx:5]: nit

```suggestion
import { type WidgetPriceIntentFragment } from '@/services/apollo/generated'
import { type Template } from '@/services/PriceCalculator/PriceCalculator.types'
```

File: apps/store/src/pages/widget/[flow]/[shopSessionId]/[priceIntentId]/calculate-price.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      variables: {
+        priceIntentId,
+      },
+    })
+
+    return data.priceIntent
+  } catch (error) {
+    console.error(`Failed to fetch price intent: ${priceIntentId}`, error)
+
+    return null
```
Comment (robinandeer) [inline apps/store/src/pages/widget/[flow]/[shopSessionId]/[priceIntentId]/calculate-price.tsx:89]: Do we need to handle this here? Why not just let it throw an error and catch in getServerSideProps?

**Review by robinandeer — APPROVED** [review]
Good start -- just left some small comments

Added via Giphy

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Nov 10, 8:20 AM**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3534).

**Pull request merged**